openwisp_users/backends.py

```diff
@@ -0,0 +1,16 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth.backends import ModelBackend
+from django.db.models import Q
+
+User = get_user_model()
+
+
+class UserAuthenticationBackend(ModelBackend):
+    def authenticate(self, request, username=None, password=None, **kwargs):
+        users = User.objects.filter(
+            Q(email=username) | Q(phone_number=username) | Q(username=username)
+        )
```

Comment:
Following up from https://github.com/openwisp/openwisp-radius/pull/207#discussion_r547051623

you could move these line into a method:

```python
def get_user(self, identifier):
    return User.objects.filter(
        Q(email=identifier) | Q(phone_number=identifier) | Q(username=identifier)
    )
```

Then in openwisp-radius you would call `auth_backend.get_user(username).first()`